File: src/api/liveCollection.ts
/**
 * Typed Tauri command wrapper for the V1AT Live Collection planning
 * surface. Planning only — no device contact, no credentials, no
 * polling, no store mutation.
 *
 * Keep names aligned with `src-tauri/src/commands/live_collection.rs`.
 */

import { invoke } from "@tauri-apps/api/core";
import type {
  LiveCollectionDryRunPlan,
  LiveCollectionDryRunRequest,
} from "../types/liveCollection";

/**
 * V1AT — return a deterministic dry-run plan for a future live
 * neighbour collection. No SSH session is opened; no command is
 * executed. The plan is for operator review only.
 */
export async function planLiveTopologyCollection(
  request: LiveCollectionDryRunRequest,
): Promise<LiveCollectionDryRunPlan> {
  return invoke<LiveCollectionDryRunPlan>(
    "plan_live_topology_collection_cmd",
    { request },
  );
}

File: src-tauri/src/commands/live_collection.rs
//! Tauri commands for the Live Collection planning surface (V1AT).
//!
//! V1AT exposes one read-only planning command. No device contact is
//! performed; the call returns a deterministic dry-run plan that an
//! operator must review before any future driver acts on it.
//!
//! See: `src-tauri/src/engines/live_collection_plan.rs` and
//! `docs/architecture/TOPOLOGY_ENGINE_BOUNDARY.md` V1AT.

use crate::engines::live_collection_plan::{
    plan_live_topology_collection, LiveCollectionDryRunPlan, LiveCollectionDryRunRequest,
};

#[tauri::command]
pub fn plan_live_topology_collection_cmd(
    request: LiveCollectionDryRunRequest,
) -> LiveCollectionDryRunPlan {
    plan_live_topology_collection(request)
}
